packages/perseus/src/interactive2/__tests__/movable.test.ts

```diff
@@ -0,0 +1,182 @@
+    });
+
+    it("modify() resets movable to initial state", () => {
+        const movable = new Movable(
+            new Graphie(document.createElement("div")),
+            {},
+        );
+
+        const state = movable.cloneState();
+        movable.state.added = false;
+
+        movable.modify();
+
+        expect(movable.cloneState()).toStrictEqual(state);
+    });
+
+    it("modify() retains original ID", () => {
+        const movable = new Movable(
+            new Graphie(document.createElement("div")),
+            {},
+        );
+        const id = movable.state.id;
+
+        movable.modify({added: true});
+
+        expect(movable.state.id).toStrictEqual(id);
+    });
+
+    it("grab() fires onMoveStart", () => {
+        const graphie = new Graphie(document.createElement("div"));
+        graphie.init({
+            range: [
+                [-10, 10],
+                [-10, 10],
+            ],
+        });
+        const onMoveStart = jest.fn();
+        const movable = new Movable(graphie, {onMoveStart});
+
+        movable.grab([0, 0]);
+
+        expect(onMoveStart).toHaveBeenCalledWith([0, 0], [0, 0]);
+    });
+
+    it("grab() fires onMove when mouse moves", () => {
+        const graphie = new Graphie(document.createElement("div"));
+        graphie.init({
+            range: [
+                [-10, 10],
+                [-10, 10],
+            ],
+        });
+        const onMove = jest.fn();
+        const movable = new Movable(graphie, {onMove});
+
+        movable.grab([0, 0]);
+        fireEvent.mouseMove(document, {clientX: 480, clientY: 320});
+
+        expect(onMove).toHaveBeenCalledWith([2, 2], [0, 0]);
+    });
+
+    it("grab() fires onMoveEnd when mouse up", () => {
+        const graphie = new Graphie(document.createElement("div"));
+        graphie.init({
+            range: [
+                [-10, 10],
+                [-10, 10],
+            ],
+        });
+        const onMoveEnd = jest.fn();
+        const movable = new Movable(graphie, {onMoveEnd});
+
+        movable.grab([0, 0]);
+        fireEvent.mouseUp(document);
+
+        expect(onMoveEnd).toHaveBeenCalledWith([0, 0], [0, 0]);
+    });
+
+    it("grab() fires onClick if still hovering when mouse up", () => {
+        const graphie = new Graphie(document.createElement("div"));
+        graphie.init({
+            range: [
+                [-10, 10],
+                [-10, 10],
+            ],
+        });
+        const onClick = jest.fn();
+        const movable = new Movable(graphie, {onClick});
+
+        movable.grab([0, 0]);
+        fireEvent.mouseUp(document);
+
+        expect(onClick).toHaveBeenCalledWith([0, 0], [0, 0]);
+    });
+
+    it("grab() does not fire onClick when mouse up if not hovering", () => {
+        const graphie = new Graphie(document.createElement("div"));
+        graphie.init({
+            range: [
+                [-10, 10],
+                [-10, 10],
+            ],
+        });
```

Comment:
Does the graphie need to be initialized for this to work?

Reply:
Good catch.  No, it passes without calling `init()`.